mullvad-relay-selector/tests/relay_selector.rs

```diff
@@ -706,6 +716,244 @@ fn test_selecting_any_relay_will_consider_multihop() {
     }
 }
 
+/// Test whether Shadowsocks is always selected as the obfuscation protocol when Shadowsocks is
+/// selected.
+#[test]
+fn test_selecting_wireguard_over_shadowsocks() {
+    let RelayListWithExtraShadowsocksIps { relay_list, .. } =
+        get_relay_list_with_extra_shadowsocks_addresses();
+
+    let relay_selector = RelaySelector::from_list(SelectorConfig::default(), relay_list);
+
+    let mut query = RelayQueryBuilder::new().wireguard().shadowsocks().build();
+    query.wireguard_constraints.use_multihop = Constraint::Only(false);
```

Comment:
Would be neat if this could be expressed by the builder using a boolean, i.e. `RelayQueryBuilder::new().wireguard().multihop(true).shadowsocks().build()`, but then I guess you would have to do the same for all settings...